platform-tests/src/test/java/org/junit/jupiter/extensions/Heavyweight.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.extensions;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolver;
+
+public class Heavyweight implements ParameterResolver {
+
+	@Override
+	public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext context) {
+		return Resource.class.equals(parameterContext.getParameter().getType());
+	}
+
+	@Override
+	public Object resolveParameter(ParameterContext parameterContext, ExtensionContext context) {
+		ExtensionContext engineContext = context.getRoot();
+		ExtensionContext.Store store = engineContext.getStore(ExtensionContext.Namespace.GLOBAL);
+		ResourceValue resource = store.getOrComputeIfAbsent(ResourceValue.class);
+		resource.usages.incrementAndGet();
+		return resource;
+	}
+
+	interface Resource extends ExtensionContext.Store.CloseableResource {
+		int usages();
+	}
+
+	static class ResourceValue implements Resource, AutoCloseable {
```

Comment:
I think we can replace `AutoCloseable` with `ExtensionContext.Store.CloseableResource` here and remove `ExtensionContext.Store.CloseableResource` on `Resource`.

Reply:
Same comment on the same line ... twice?

Reply:
Ask GitHub... ;-)